uaclient/entitlements/landscape.py

```diff
@@ -0,0 +1,166 @@
+import logging
+import os
+import subprocess
+from typing import Any, Dict, Optional, Tuple
+
+from uaclient import apt, event_logger, exceptions, messages, system, util
+from uaclient.entitlements.base import UAEntitlement
+from uaclient.entitlements.entitlement_status import ApplicationStatus
+
+event = event_logger.get_event_logger()
+
+LANDSCAPE_CLIENT_PACKAGE_NAME = "landscape-client"
+LANDSCAPE_CLIENT_CONFIG_PATH = "/etc/landscape/client.conf"
+LANDSCAPE_CLIENT_CONFIG_PATH_DISABLE_BACKUP = (
+    "/etc/landscape/client.conf.pro-disable-backup"
+)
+
+
+class LandscapeEntitlement(UAEntitlement):
+    name = "landscape"
+    title = "Landscape"
+    description = "Management and administration tool for Ubuntu"
+    help_doc_url = "https://ubuntu.com/landscape"
+
+    def _perform_enable(self, silent: bool = False) -> bool:
+        event.info(messages.APT_UPDATING_LISTS)
+        apt.run_apt_update_command()
+
+        event.info(
+            messages.INSTALLING_PACKAGE.format(LANDSCAPE_CLIENT_PACKAGE_NAME)
+        )
+        apt.run_apt_install_command([LANDSCAPE_CLIENT_PACKAGE_NAME])
+
+        cmd = ["landscape-config"] + self.extra_args
+        stdout = None
+        stderr = None
+        if self.assume_yes:
+            if "--silent" not in cmd:
+                cmd += ["--silent"]
+            stdout = subprocess.PIPE
+            stderr = subprocess.PIPE
+
+        logging.debug(messages.EXECUTING_COMMAND.format(" ".join(cmd)))
+        event.info(
+            util.redact_sensitive_logs(
+                messages.EXECUTING_COMMAND.format(" ".join(cmd))
+            )
+        )
+        proc = subprocess.run(cmd, stdout=stdout, stderr=stderr)
```

Comment:
why can't we use `subp` here ?

Reply:
`subp` captures stdout and stderr. In the interactive case for landscape, we want `landscape-config` to inherit stdout and stderr and work interactively.
It seemed simpler to make this a special case rather than build support for not capturing stdout/stderr in the shared `subp` function

Reply:
Updated